packages/lib/src/components/Card/components/CardInput/CardInput.tsx

```diff
@@ -87,10 +89,23 @@ class CardInput extends Component<CardInputProps, CardInputState> {
     }
 
     componentDidUpdate(prevProps, prevState) {
-        const { country: prevCountry, stateOrProvince: prevStateOrProvince } = prevState.billingAddress || {};
-        const { country, stateOrProvince } = this.state.billingAddress || {};
-
-        if (prevCountry !== country || prevStateOrProvince !== stateOrProvince) {
+        /**
+         * Validating every time there's a change in state
+         */
+        if (
+            !objectsDeepEqual(prevState.billingAddress, this.state.billingAddress) ||
+            prevState.storePaymentMethod !== this.state.storePaymentMethod ||
+            !objectsDeepEqual(prevState.installments, this.state.installments) ||
+            prevState.isSfpValid !== this.state.isSfpValid ||
+            prevState.hideCVCForBrand !== this.state.hideCVCForBrand ||
+            prevState.brand !== this.state.brand ||
+            prevState.additionalSelectValue !== this.state.additionalSelectValue ||
+            // Covers changes to:
+            // - encryptedField values
+            // - KCP authentication
+            // - holder name
+            !objectsDeepEqual(prevState.data, this.state.data)
```

Comment:
Maybe this could be extracted into a function

Reply:
Why?

Reply:
I can - if it will make you happy ;-) What shall I call it. `checkStateToSeeWhetherToValidateCardInput` :-)

Reply:
Just thinking it could maybe make the `CardInput.tsx` file a bit more readable, as you prefer! 👍